htlcswitch/hop/iterator.go

```diff
@@ -251,55 +251,55 @@ func (p *OnionProcessor) DecodeHopIterators(id []byte,
 
 	tx := p.router.BeginTxn(id, batchSize)
 
-	for i, req := range reqs {
-		onionPkt := &onionPkts[i]
-		resp := &resps[i]
+	decode := func(seqNum uint16, onionPkt *sphinx.OnionPacket,
```

Comment:
Since you've added a separate commit to separate this out, does this need to be declared inline?
Could be cleaner to have a decode fn that returns `FailCode, err`?

Reply:
I remember I did try that, but it didn't become much better with the (return) parameters.